src/Workspaces/VisualBasic/Portable/Extensions/SyntaxTokenExtensions.vb

```diff
@@ -57,6 +57,31 @@ Namespace Microsoft.CodeAnalysis.VisualBasic.Extensions
         End Function
 
         <Extension()>
+        Public Function IsGlobalStatementContext(token As SyntaxToken, position As Integer) As Boolean
+            If Not token.IsLastTokenOfStatement() Then
+                Return False
+            End If
+
+            ' NB: Checks whether the caret is placed after a colon or an end of line.
+            ' Otherwise the typed expression would still be a part of the previous statement.
+            If Not token.HasTrailingTrivia Then
+                Return False
+            End If
+
+            For Each trivia In token.TrailingTrivia
+                If trivia.Span.Start > position Then
+                    Return False
+                ElseIf trivia.IsKind(SyntaxKind.ColonTrivia) Then
+                    Return True
```

Comment:
The recommendation service already checks if the caret is not inside of a comment.
Also the `token` passed is the closest token to the left of the caret. The trivia between the caret and the token should contain just whitespace.